File: employee-management-app/src/app/pages/profile/models/userSkills.ts
import {UserProfile} from "./userProfile";
import {Project} from "./project";

export class UserSkills {
  id: string;
  user: UserProfile | undefined;
  name: string;
  rating: number;

  constructor(
    id: string,
    user: UserProfile | undefined,
    name: string,
    rating: number
  ) {
    this.id = id;
    this.user = user;
    this.name = name;
    this.rating = rating;
  }
}

File: employee-management/src/main/java/com/dreamteam/employeemanagement/model/UserSkills.java
package com.dreamteam.employeemanagement.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.UUID;
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "skills")
public class UserSkills {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String name;
    private Double rating;
    @ManyToOne
    @JoinColumn(name = "userId")
    private Account user;

    public String getId() {
        return id.toString();
    }
}
